### ISSUE TITLE
CSS in JS

### ISSUE THREAD

**Issue by troutowicz** [troutowicz/geoshare#11]
Let's make geoshare 100% JS. @vjeux's [presentation](https://speakerdeck.com/vjeux/react-css-in-js) gives an overview of several well known CSS problems and how inline component styling makes these obsolete and just makes more sense from a component perspective.

See [css-in-js](https://github.com/troutowicz/geoshare/tree/css-in-js) branch for current WIP.

**Comment by ababol** [comment]
Hi @troutowicz, great job! Your project is awesome. :tada: 

You probably already know [react-style](https://github.com/js-next/react-style), have you considered using it?

**Comment by troutowicz** [comment]
Thanks @ababol! :beers:

I do know about react-style. The reason geoshare is not using it is simply because the majority of geoshare's components use [material-ui](https://github.com/callemall/material-ui) components and that library is currently using their own implementation of handling inline styling. react-style is a nice approach though no doubt, I really like the generated class names and css extraction ability with Webpack.

Edit: I guess geoshare could still use react-style, I was stuck on the fact that material-ui does things differently... but that doesn't mean I have to follow suit. :)

**Comment by ababol** [comment]
@troutowicz Ok thanks for the answer!

Yep, so I forked locally your repository yesterday to make a Meerkat live map (instead of Instagram).
I have started to make my new components by using the material-ui approach of inline styling but I am definitely more comfortable with the react-style approach, it's really easy to use and I also like the fact that it follows the react-native inline styling approach. If one day I want to make a react-native application by using the same components, it would be easier.
